### PR TITLE
Added functions to part configuration

### PR DESCRIPTION
**Pull request by lgiordani**
This set of commits extracts the `numeric` and `values` configuration options as classes that are activated through the `function` configuration key.

I also added a `date` function for date-based version parts.

The documentations has been updated accordingly

### CODE REVIEW

File: bumpversion/__init__.py [hunk 1]
```diff
@@ -340,7 +340,7 @@ class VersionConfig(object):
     Holds a complete representation of a version string
     """
 
-    def __init__(self, parse, serialize, search, replace, part_configs=None):
+    def __init__(self, parse, serialize, search, replace, part_configs={}):
```
Comment (peritus) [inline bumpversion/__init__.py]: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

**Comment by peritus** [comment]
:+1:

File: bumpversion/__init__.py (hunk 1 above)
Comment (lgiordani) [inline bumpversion/__init__.py]: Really the most stupid error in Python. Sorry =) =)

Comment (peritus) [inline bumpversion/__init__.py]: That's what code review is for, finding the oversights that are obvious if you're not the person coding.

**Comment by ryneeverett** [comment]
Why closed?

**Comment by lgiordani** [comment]
This has been opened one year ago, and was an attempt to help contributing to a bumpversion refactoring. Since bumpversion is no more actively developed I implemented some ideas in the punch package, and no more maintaining my bumpversion fork. If @peritus is still interested in it the commits can be still retrieved.

**Comment by haidaraM** [comment]
Looks like this pull request solves this #144

**Pull request closed without merging**